### PR TITLE
Update logging config

### PR DESCRIPTION
**Pull request by peytondmurray**
## Why are these changes needed?

This PR simplifies some logging configuration in accordance with extended discussions with @xwjiang2010 and @rkooo567. Changes:

- Avoid an unnecessary binding the 'filelock' logger to a variable in `python/ray/autoscaler/_private/local/node_provider.py` to discourage use within ray
- Made a note in `logservicer.py` (mostly for myself) that the logging configuration there is not to be messed with because it runs in a separate thread
- Log to a module logger `release/ray_release/logger.py`, `rllib/examples/sumo_env_local.py`, and `rllib/examples/tune/framework.py`, `python/ray/_private/runtime_env/_clonevirtualenv.py`
- Remove custom logging configuration in `rllib` and `experimental/raysort/main.py`; we will use the global ray logging configuration here moving forward
- Logging configuration in `rllib` can no longer be set from `AlgorithmConfig`; use standard logging configuration routes going forward
- Added an explanation of logging configuration in kuberay

Anything that runs in a separate process was left unmodified, as per the discussion here: #30005.

## Related issue number

Partly addresses #30005.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/kuberay/utils.py [hunk 1]
```diff
@@ -33,6 +33,18 @@
 
 
 def setup_logging():
+    """Set up logging for kuberay.
+
+    For kuberay's autoscaler integration, the autoscaler runs in a sidecar container
```
Comment (rkooo567) [inline python/ray/tests/kuberay/utils.py:38]: Consider removing this docstring? When I look at the implementation, This docstring doesn't make much sense to me

File: python/ray/autoscaler/_private/kuberay/run_autoscaler.py [hunk 2]
```diff
@@ -17,7 +17,21 @@
 
 
 def run_kuberay_autoscaler(cluster_name: str, cluster_namespace: str):
-    """Wait until the Ray head container is ready. Then start the autoscaler."""
+    """Wait until the Ray head container is ready. Then start the autoscaler.
+
+    For kuberay's autoscaler integration, the autoscaler runs in a sidecar container
```
Comment (rkooo567) [inline python/ray/autoscaler/_private/kuberay/run_autoscaler.py:22]: can we just move this to line 44? Docstring should explain the API (`run_kuberay_autoscaler `), but this string looks like an implementation detail

**Review by rkooo567 — COMMENTED** [review]
For rllib changes, I think we need rllib folks approval. cc @sven1977 who's the best person who can take a look at this PR?

File: python/ray/tests/kuberay/utils.py (hunk 1 above)
Comment (DmitriGekhtman) [reply python/ray/tests/kuberay/utils.py:38]: This function just sets up logging for the test -- the doc string is inaccurate.

Comment (peytondmurray) [reply python/ray/tests/kuberay/utils.py]: Whoops, I think I copied the docstring from the other location. Removing now.

File: python/ray/autoscaler/_private/kuberay/run_autoscaler.py (hunk 2 above)
Comment (peytondmurray) [reply python/ray/autoscaler/_private/kuberay/run_autoscaler.py]: This has been moved to where `_setup_logging()` is called further down.

File: python/ray/autoscaler/_private/kuberay/run_autoscaler.py [hunk 3]
```diff
@@ -41,8 +41,14 @@ def run_kuberay_autoscaler(cluster_name: str, cluster_namespace: str):
...
 
-    # The Ray head container sets up the log directory. Thus, we set up logging
-    # only after the Ray head is ready.
+    # For kuberay's autoscaler integration, the autoscaler runs in a sidecar container
+    # in the same pod as the main Ray container, which runs the rest of the Ray
+    # processes.
+    # The logging configuration here is for the sidecar container, but we need the
+    # logs to go to the same place as the head node logs because the autoscaler
+    # is allowed to communicate with another Ray process (the log monitor) via logs
+    # in that directory. However, the Ray head container sets up the log directory.
```
Comment (DmitriGekhtman) [inline python/ray/autoscaler/_private/kuberay/run_autoscaler.py:50]: Could you split this sentence up?

Comment (peytondmurray) [reply python/ray/autoscaler/_private/kuberay/run_autoscaler.py]: Definitely. I've split it up and reworded it - what do you think now?

**Review by DmitriGekhtman — APPROVED** [review]

File: python/ray/autoscaler/_private/kuberay/run_autoscaler.py (hunk 3 above)
Comment (DmitriGekhtman) [reply python/ray/autoscaler/_private/kuberay/run_autoscaler.py]: Looks good, thanks!

**Review by rkooo567 — APPROVED** [review]
LGTM though we still need approval from the rllib. I will ping one of guys there

**Comment by rkooo567** [comment]
can you also merge the latest master?

**Comment by rkooo567** [comment]
cc @gjoliver to take a look

**Review by sven1977 — COMMENTED** [review]
Thanks for this cleanup @peytondmurray ! One more thing for RLlib. Since we don't use the `AlgorithmConfig.log_level` setting anymore, we should properly deprecate it.

In `rllib/algorithms/algorithm_config.py`, can you:

```
self.log_level = "WARN" -> self.log_level = DERPECATED_VALUE

...
def debugging()
...
        log_level: Optional[str] = NotProvided, -> log_level=DEPRECATED_VALUE,

...
    # and further down in that same method:
    if log_level != DERECATED_VALUE:
        deprecation_warning(old="config.log_level", help="[describe here, how RLlib users should set their log levels, instead]", error=False)
```

Thanks so much!

**Comment by peytondmurray** [comment]
Thanks @sven1977, I just pushed your requested changes. If the tests pass :crossed_fingers: I think we'll be good to merge.

**Comment by rkooo567** [comment]
@sven1977 we need your approval for the merge! Thank you!

**Comment by rkooo567** [comment]
cc @gjoliver @sven1977

**Review by sven1977 — APPROVED** [review]
LGTM. Thanks for this PR @peytondmurray !

**Pull request merged**